examples/gno.land/r/gnoland/boards2/v1/README.md

```diff
@@ -0,0 +1,526 @@
+- Ability to submit `Call` transactions with appropriate gas/fees
+
+### Basic Operations
+
+1. **Discover or create a board** (requires realm membership for creation)
+   - Resolve by name: `GetBoardIDFromName(name)`
+   - Or create: `CreateBoard(name, listed)` (requires `board:create` permission)
+
+2. **Post content**
+   - Create a thread: `CreateThread(boardID, title, body)`
+   - Reply to a thread: `CreateReply(boardID, threadID, 0, body)`
+
+3. **Moderate**
+   - Flag problematic content: `FlagThread(...)` / `FlagReply(...)`
+   - Freeze to pause activity: `FreezeBoard(...)`, `FreezeThread(...)`
+   - Ban repeat offenders: `Ban(boardID, user, hours, reason)`
+
+4. **Manage members**
+   - Invite or accept requests: `InviteMember(...)`, `AcceptInvite(...)`
+   - Change roles: `ChangeMemberRole(...)`
+
+5. **Govern**
+   - Adjust thresholds: `SetFlaggingThreshold(boardID, n)`
+   - Lock realm (irreversible): `LockRealm(lockRealmMembers)`
+
+## Overview
+
+Boards operates on two permission scopes that are fundamental to understand:
+
+**Realm-level operations** (`boardID=0`): Control who can create boards, manage realm members, set notices, and lock the realm. Uses `gPerms` (global realm permissions).
+
+**Board-level operations** (`boardID!=0`): Control activities within a specific board - posting, moderation, member management for that board. Each board has its own `Permissions` instance.
+
+This dual-scope design allows realm governance (who can create boards) to be separate from board governance (how individual boards operate).
+
+- Boards organizes content into boards → threads → replies
+- Each board has its own permissions (roles, thresholds, ban-list, freeze state)
+- Realm-level governance can be locked permanently and independently from member modifications
+- Moderation leverages per-board flagging thresholds, freezing, and bans
+- Permissions are enforced via a pluggable `Permissions` interface; the default implementation is `BasicPermissions` backed by a DAO member set
+
+### Key Packages and Files
+
+- Public API: `examples/gno.land/r/gnoland/boards2/v1/public.gno`
+- Board model and helpers: `examples/gno.land/r/gnoland/boards2/v1/board.gno`
+- Post model and helpers: `examples/gno.land/r/gnoland/boards2/v1/post.gno`
+- Permissions: `examples/gno.land/r/gnoland/boards2/v1/permissions.gno`, `permissions_basic.gno`, `permissions_validators.gno`
+- Moderation (flag/ban/freeze): `public_flag.gno`, `flag.gno`, `public_ban.gno`, `public_freeze.gno`
+- Realm lock: `public_lock.gno`
+- Invitations: `public_invite.gno`
+- Rendering helpers/URIs: `render.gno`, `uris_post.gno`
+
+## Roles and Permissions
+
+Authorization model controls who can do what, and where. Boards uses a pluggable permission engine to enforce capabilities at realm and board scopes. Use the default mapping, refine with validators, or swap the engine with `SetPermissions` to fit governance.
```

Comment:
Maybe?

```suggestion
Authorization model controls who can do what, and where. Boards uses a pluggable permission engine to enforce capabilities at realm and board scopes. When the default engine is not adequate it can be swapped to a custom one with `SetPermissions(...)` to fit governance.
```